### PR TITLE
Feat: Static Resolver implementation

### PR DESCRIPTION
**Pull request by cbrzn**

### CODE REVIEW

File: packages/polywrap-client/polywrap_client/client.py [hunk 1]
```diff
@@ -86,7 +83,7 @@ async def try_resolve_uri(
         self, options: TryResolveUriOptions
     ) -> Result[UriPackageOrWrapper]:
         uri = options.uri
-        uri_resolver = self._config.resolver
+        uri_resolver = self.get_uri_resolver()
```
Comment (Niraj-Kamdar) [inline packages/polywrap-client/polywrap_client/client.py:86]: Why this change? It's alright to access the config attribute in a method

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from abc import ABC, abstractmethod
+
+from polywrap_core import IUriResolver, Uri, IUriResolutionContext, UriPackageOrWrapper, Client, IUriResolutionStep
+from polywrap_result import Result
+
+
+class ResolverWithHistory(IUriResolver, ABC):
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py:7]: Shouldn't this be IResolverWithHistory?

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        resolution_context.track_step(step)
+
+        return result
+
+    @abstractmethod
+    def get_step_description(self) -> str:
+        pass
+
+    @abstractmethod
+    async def _try_resolve_uri(self, uri: Uri, client: Client, resolution_context: IUriResolutionContext) -> Result["UriPackageOrWrapper"]:
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py:24]: Why is this an abstract method? This should be optional and upto implementation of interface top define it or not.

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/aggregator/uri_resolver_aggregator.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from .uri_resolver_aggregator_base import UriResolverAggregatorBase
+
+class UriResolverAggregator(UriResolverAggregatorBase):
+    resolvers: List[IUriResolver]
+    name: Optional[str]
+
+    def __init__(self, resolvers: List[IUriResolver]):
+      self.resolvers = resolvers
+
+    def get_step_description(self) -> str:
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/aggregator/uri_resolver_aggregator.py:15]: Shouldn't this be get_step_name?

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/legacy/base_resolver.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-from typing import Dict, List
+from typing import Dict
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/legacy/base_resolver.py:1]: Wyhy do we still have these legacy resolvers?

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class StaticResolver(IUriResolver):
+    uri_map: dict[str, UriPackageOrWrapper]
+
+    def __init__(self, uri_map: dict[str, UriPackageOrWrapper]):
+        self.uri_map = uri_map
+
+    @staticmethod
+    def _from(static_resolver_likes: List[UriResolverLike]) -> "StaticResolver":
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:15]: This should be class method instead. Let's name this something else. (Ex: construct_from or from_list, from_array?)

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        uri_map: dict[str, UriPackageOrWrapper] = dict()
+        for static_resolver_like in static_resolver_likes:
+            if type(static_resolver_like) == list:
+                resolver = StaticResolver._from(cast(List[UriResolverLike], static_resolver_like))
+                for uri, package_or_wrapper in resolver.uri_map.items():
+                    uri_map[uri] = package_or_wrapper
+
+            elif hasattr(static_resolver_like, "uri") and hasattr(static_resolver_like, "package"):
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:23]: There should be better way to check this with isinstance

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    uri_map: dict[str, UriPackageOrWrapper]
+
+    def __init__(self, uri_map: dict[str, UriPackageOrWrapper]):
+        self.uri_map = uri_map
+
+    @staticmethod
+    def _from(static_resolver_likes: List[UriResolverLike]) -> "StaticResolver":
+        uri_map: dict[str, UriPackageOrWrapper] = dict()
+        for static_resolver_like in static_resolver_likes:
+            if type(static_resolver_like) == list:
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:18]: use isinstance here

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+                    uri_map[uri] = package_or_wrapper
+
+            elif hasattr(static_resolver_like, "uri") and hasattr(static_resolver_like, "package"):
+                uri_package = UriPackage(uri=static_resolver_like.uri, package=static_resolver_like.package) # type: ignore
+                uri_map[uri_package.uri.uri] = uri_package
+            elif hasattr(static_resolver_like, "uri") and hasattr(static_resolver_like, "wrapper"):
+                uri_wrapper = UriWrapper(uri=static_resolver_like.uri, wrapper=static_resolver_like.wrapper) # type: ignore
+                uri_map[uri_wrapper.uri.uri] = uri_wrapper
+            else:
+                raise Exception("Unknown static-resolver-like type provided.")
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:30]: Why raise use Result here

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        return StaticResolver(uri_map)
+
+    async def try_resolve_uri(self, uri: Uri, client: Client, resolution_context: IUriResolutionContext) -> Result["UriPackageOrWrapper"]:
+        package_or_wrapper = self.uri_map.get(uri.uri)
+
+        result: Result[UriPackageOrWrapper] = UriResolutionResult.ok(uri)
+        description: str = "StaticResolver - Miss"
+
+        if package_or_wrapper:
+            if hasattr(package_or_wrapper, "package"):
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:41]: use isinstance instead.

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/uri_resolver_wrapper.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+from polywrap_result import Result, Ok, Err
+
+class UriResolverWrapper(ResolverWithHistory):
+    implementation_uri: Uri
+
+    def __init__(self, uri: Uri) -> None:
+        self.implementation_uri = uri
+
+    def get_step_description(self) -> str:
+        return super().get_step_description()
```
Comment (Niraj-Kamdar) [inline packages/polywrap-uri-resolvers/polywrap_uri_resolvers/uri_resolver_wrapper.py:13]: Isn't this abstract? Why call super?

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]
Left some comments regarding better code practices

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py (hunk 3 above)
Comment (Niraj-Kamdar) [reply packages/polywrap-uri-resolvers/polywrap_uri_resolvers/abc/resolver_with_history.py:24]: NVM I got this

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/aggregator/uri_resolver_aggregator.py (hunk 4 above)
Comment (cbrzn) [reply packages/polywrap-uri-resolvers/polywrap_uri_resolvers/aggregator/uri_resolver_aggregator.py:15]: we are following the `UriResolverAggregatorBase` interface and I think it makes sense to keep it with description

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py (hunk 6 above)
Comment (cbrzn) [reply packages/polywrap-uri-resolvers/polywrap_uri_resolvers/static_resolver.py:15]: I think it make sense to keep it as a static method, but the name indeed need to be changed; will rename it to `from_list`

File: packages/polywrap-uri-resolvers/polywrap_uri_resolvers/uri_resolver_wrapper.py (hunk 11 above)
Comment (cbrzn) [reply packages/polywrap-uri-resolvers/polywrap_uri_resolvers/uri_resolver_wrapper.py:13]: this was autogenerated but the code editor, my bad

Comment (cbrzn) [reply packages/polywrap-uri-resolvers/polywrap_uri_resolvers/uri_resolver_wrapper.py:13]: actually i will just update this with returning a string and then, when the plugins are updated and this resolver is developed I will fix this
